### PR TITLE
Add plot module

### PR DESCRIPTION
**Pull request by cgohlke**
## Description

This PR adds a ``plot`` module with functions and classes to create publication quality visualizations of phasor coordinates and related data using the matplotlib library.

The following tutorials are added or updated:

- Phasor plot: An introduction to the `PhasorPlot` class.
- Phasor coordinates from lifetimes: An introduction to the `phasor_from_lifetime` function.
- Introduction to PhasorPy: An introduction to using the PhasorPy library.

The introduction tutorial depends on [scikit-image](https://github.com/scikit-image/scikit-image) for median filter.

## Release note

Summarize the changes in the code block below to be included in the
[release notes](https://www.phasorpy.org/stable/release.html):

```release-note
Add plot module
```

## Checklist

- [x] The pull request title, summary, and description are concise.
- [x] Related issues are linked in the description.
- [x] New dependencies are explained.
- [x] The source code and documentation can be distributed under the [MIT license](https://www.phasorpy.org/stable/license.html).
- [x] The source code adheres to [code standards](https://www.phasorpy.org/stable/contributing.html#code-standards).
- [x] New classes, functions, and features are thoroughly [tested](https://www.phasorpy.org/stable/contributing.html#tests).
- [x] New, user-facing classes, functions, and features are [documented](https://www.phasorpy.org/stable/contributing.html#documentation).
- [x] New features are covered in tutorials.
- [x] No files other than source code, documentation, and project settings are added to the repository.

### CODE REVIEW

**Comment by cgohlke** [comment]
This PR is ready for review. 

Since the discussion last week, I improved the universal semicircle, added convenience functions for plotting signal and phasor images, and finished documentations and tests.

The `contour` and `imshow` methods are not implemented yet. This PR is already large enough.

It would be nice to merge #37 before this so the introduction tutorial can use the higher level `phasor_calibrate` function.

**Review by bruno-pannunzio — APPROVED** [review]
Hi @cgohlke. I was thinking that it will take some days until I can finish all tests for #37. Maybe we can merge this PR if it is ready, which I think it is, and then I can modify the introduction tutorial to use the higher level function instead. What do you think?

**Comment by cgohlke** [comment]
> Maybe we can merge this PR if it is ready

Sure. You could also update the tutorial in #37 after this is merged, or it can be updated later.

**Comment by schutyb** [comment]
I tried the full circle for spectral it and for me its OK, just a little thing in the test_plot.py file the class **TestPhasoPlot:**  should not be **TestPhasorPlot:**  with an **r**

    """Test the PhasorPlot class."""

File: tests/test_plot.py
Comment (schutyb) [inline tests/test_plot.py:21]: TestPhasorPlot

**Review by schutyb — APPROVED** [review]

**Comment by cgohlke** [comment]
Thanks @bruno-pannunzio and @schutyb. I pushed some minor fixes and will merge this PR.

**Comment by cgohlke** [comment]
The updated tutorials are now online @ https://www.phasorpy.org/stable/tutorials/

**Pull request merged**
